executors/caiexec/volume.go

```diff
@@ -0,0 +1,314 @@
+	// 	UsedSpace        int64    `xml:"used_space"`
+	// 	AvailSpace       int64    `xml:"avail_space"`
+	// 	SlExceeded       string   `xml:"sl_exceeded"`
+	// 	HlExceeded       string   `xml:"hl_exceeded"`
+	// }
+	//
+	// type VolQuota struct {
+	// 	XMLName xml.Name `xml:"volQuota"`
+	// 	Limits  []Limit  `xml:"volQuota"`
+	// }
+	//
+	// command := []string{
+	// 	fmt.Sprintf("gluster --mode=script volume quota %v list --xml", volume.Name),
+	// }
+	//
+	// // Get the xml output of volume quota list
+	// output, err := s.RemoteExecutor.RemoteCommandExecute(host, command, 10)
+	// if err != nil {
+	// 	return nil, fmt.Errorf("Unable to get volume quota list of volume name: %v", volume)
+	// }
+	// var volQuota VolQuota
+	// err = xml.Unmarshal([]byte(output[0]), &volQuota)
+	// if err != nil {
+	// 	return nil, fmt.Errorf("Unable to determine volume quota list of volume name: %v", volume)
+	// }
+	// logger.Debug("%+v\n", volQuota)
+	// var currentQuota Limit
+	//
+	// for _, limit := range volQuota.Limits {
+	// 	if limit.Path == "/" {
+	// 		currentQuota = limit
+	// 	}
+	// }
+
+	// TBD: simply expand with expected limit
+	command := []string{
+		fmt.Sprintf("gluster --mode=script volume quota %v limit-usage / %vGB", volume.Name, volume.Size),
+	}
+
+	_, err := s.RemoteExecutor.RemoteCommandExecute(host, command, 10)
+	if err != nil {
+		return nil, err
+	}
+
+	return &executors.Volume{}, nil
+}
+
+func (s *CaiExecutor) VolumeDestroy(host string, volume string) error {
+	godbc.Require(host != "")
+	godbc.Require(volume != "")
+
+	// First stop the volume, then delete it
+
+	commands := []string{
+		fmt.Sprintf("gluster --mode=script volume stop %v force", volume),
+	}
+
+	_, err := s.RemoteExecutor.RemoteCommandExecute(host, commands, 10)
+	if err != nil {
+		logger.LogError("Unable to stop volume %v: %v", volume, err)
+	}
+
+	commands = []string{
+		fmt.Sprintf("gluster --mode=script volume delete %v", volume),
+	}
+
+	_, err = s.RemoteExecutor.RemoteCommandExecute(host, commands, 10)
+	if err != nil {
+		return logger.Err(fmt.Errorf("Unable to delete volume %v: %v", volume, err))
+	}
+
+	return nil
+}
+
+func (s *CaiExecutor) VolumeDestroyCheck(host, volume string) error {
+	godbc.Require(host != "")
+	godbc.Require(volume != "")
+
+	// Determine if the volume is able to be deleted
+	err := s.checkForSnapshots(host, volume)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *CaiExecutor) createVolumeOptionsCommand(volume *executors.VolumeRequest) []string {
+	commands := []string{}
+	var cmd string
+
+	// Go through all the Options and create volume set command
+	for _, volOption := range volume.GlusterVolumeOptions {
+		if volOption != "" {
+			cmd = fmt.Sprintf("gluster --mode=script volume set %v %v", volume.Name, volOption)
+			commands = append(commands, cmd)
+		}
+
+	}
+	return commands
+}
+
+func (s *CaiExecutor) createQuotaCommand(volume *executors.VolumeRequest) []string {
```

Comment:
Since we are not enforcing on directory (only on volume), I think we can just ignore it.